src/common/libsubprocess/subprocess.c

```diff
@@ -1121,6 +1121,21 @@ int flux_subprocess_fail_errno (flux_subprocess_t *p)
     return p->failed_errno;
 }
 
+const char *flux_subprocess_fail_error (flux_subprocess_t *p)
+{
+    if (!p) {
+        errno = EINVAL;
+        return NULL;
+    }
+    if (p->state != FLUX_SUBPROCESS_FAILED) {
+        errno = EINVAL;
+        return NULL;
+    }
+    if (p->failed_error.text[0] == '\0')
+        return strerror (p->failed_errno);
+    return p->failed_error.text;
+}
```

Comment:
I'm not sure I know all the places where this will be used but sometimes it's inconvenient to return NULL from a function like this.  Would it be better to to just return textual errors like
- `internal error: subprocess is NULL`
- `internal error: subprocess is not in FAILED state`

then the user can just use the result in a printf argument without checking?